### PR TITLE
Make the NUOPC driver the default

### LINKED ISSUE #1083: NUOPC cap: use named constants rather than literal strings in state_getimport and state_setexport
I was looking quickly at the NUOPC cap. It looks like `state_getimport` and `state_setexport` return silently if the given field doesn't exist. I feel that this behavior, combined with the use of strings (rather than defined constants that can be checked at compile time) is an opening for bugs due to typos.

I feel it would be more robust to make these routines abort if the given field isn't found. I realize that there may then need to be some logic around the calls to `state_getimport` and `state_getexport` so that they aren't called if a given field is known not to exist.

For example, this line would probably then result in an error, because methane currently isn't added to the field list:

```Fortran
call state_getimport(importState, 'Sa_methane', bounds, output=atm2lnd_inst%forc_pch4_grc, rc=rc )
```

Alternatively, the use of strings throughout the cap could be replaced with constants. For example, the use of `Faxa_snowc` in these lines:

```Fortran
    call fldlist_add(fldsToLnd_num, fldsToLnd, 'Faxa_snowc'   )
```
```Fortran
    call state_getimport(importState, 'Faxa_snowc', bounds, output=forc_snowc, rc=rc )
```

would be replaced by reference to a named constant (whose value could still be `Faxa_snowc`). This would ensure that you don't try to reference a non-existent name in the `state_getimport` call.

@mvertens I'm tentatively assigning you. Feel free to comment if you feel my suggestions aren't feasible.

### LINKED ISSUE #1192: Incompatible initial conditions for some f09 Hist cases
### Brief summary of bug

I ran the `ctsm_sci` test list on a branch off of `ctsm5.1.dev007`. A number of the f09 Hist cases failed, seemingly due to incompatible finidat files.

### General bug information

**CTSM version you are using:** `ctsm5.1.dev007-16-gd44929a7f`. I don't think anything relevant has changed in this tag relative to ctsm5.1.dev007. I haven't checked if this is fixed in dev008, though based on the ChangeLog, it doesn't look like it.

**Does this bug cause significantly incorrect results in the model's science?** No

**Configurations affected:** Observed for a variety of Hist compsets at f09 resolution:
```
    FAIL SMS_Ld5.f09_g17.IHistClm45BgcCruGs.cheyenne_intel.clm-default RUN time=23
    FAIL SMS_Ld5.f09_g17.IHistClm45BgcGs.cheyenne_intel.clm-default RUN time=23
    FAIL SMS_Ld5.f09_g17.IHistClm50Sp.cheyenne_intel.clm-default RUN time=23
    FAIL SMS_Ld5.f09_g17.IHistClm50SpCru.cheyenne_intel.clm-default RUN time=24
```

Here are the f09 tests that passed:

```
    PASS SMS_Ld1.f09_g17.I1850Clm50Sp.cheyenne_intel.clm-default RUN time=22
    PASS SMS_Ld1.f09_g17.I1850Clm50SpCru.cheyenne_intel.clm-default RUN time=22
    PASS SMS_Ld3.f09_g17.I1850Clm50BgcCropCru.cheyenne_intel.clm-default RUN time=33
    PASS SMS_Ld5.f09_g17.I1850Clm45BgcCruGs.cheyenne_intel.clm-default RUN time=35
    PASS SMS_Ld5.f09_g17.I1850Clm45BgcGs.cheyenne_intel.clm-default RUN time=38
    PASS SMS_Ld5.f09_g17.I1850Clm50Bgc.cheyenne_intel.clm-default RUN time=67
    PASS SMS_Ld5.f09_g17.IHistClm45SpGs.cheyenne_intel.clm-default RUN time=50
    PASS SMS_Ld5.f09_g17.IHistClm50BgcCrop.cheyenne_intel.clm-default RUN time=98
```

### Details of bug

I just checked one of the above failures and the issue was that init_interp seemed to be needed, but use_init_interp wasn't set for the case. I should note that my testing was done from a branch that I'm using to test pio2, but I don't think anything relevant has changed on this branch, so I assume the same problem exists on master.

@ekluzek the new ctsm_sci test list seems valuable in picking up issues like this. I wonder if we should have a rule that we run this test list (without baseline comparisons) whenever we remake many of the initial conditions and/or surface datasets?

### LINKED ISSUE #1237: For nuopc cap: add a namelist flag, force_send_to_atm
@mvertens has implemented some code in the nuopc cap that avoids sending lnd -> atm fields to the mediator if the atmosphere isn't prognostic. This is helpful for performance, but is problematic for testing, where we want to be able to confirm that we haven't changed any lnd -> atm fields despite running our tests with datm.

I proposed, and @mvertens agreed, that we should handle this by having a namelist flag like `force_send_to_atm`. Our default testmod (as well as other testmods that don't inherit from default) can set this to true, so that, for testing, we'll send lnd -> atm fields even if running with datm, so that they will be output to the cpl history files. However, at least once we switch to mainly running our tests with nuopc, we should have at least one test that *does not* set this, so that we have a test of the standard production configuration, to make sure that works. (As a general rule, we should have at least one test that doesn't include test-specific options, such as those included in the `default` testmod.)

For now, I'm keeping @mvertens 's code in place, but I'm putting in a kludge to act as if such a namelist flag is always present and set to true.

### LINKED ISSUE #1365: NUOPC changes DATM_CLMNCEP_YR_START/END to DATM_YR_START/END
This appears to just be a problem in some of the tools, documentation, the User's Guide and the USUMB test case. But between the MCT and NUOPC data models the DATM_CLMNCEP_YR xml variables were renamed to remove the CLMNCEP part of it. xmlchange shows an error when you try to change something that doesn't exist, so you have to know which one is being used in order to run. So tools or user-mods that assume one or the other have to know which driver is being used.

We'll need a testmod for USUMB that's specific for NUOPC, or it needs to check the driver being used.

The tools:

tools/contrib/run_clm_historical
tools/contrib/run_clmtowers

should either change to assume NUOPC (once we make it the default) or have a switch for the coupler being used. I suppose we should just change the User's Guide when NUOPC is the default.

FYI: @jedwards4b @mvertens

### LINKED ISSUE #1476: Remove blank files from ctsm.input_data_list
We use a " " (i.e. blank) filename to declare that a file is not used in some cases (finidat, and landuse.timeseries for example), these still get added to the ctsm.input_data_list and cime therefore writes out a warning about them when check_input_data is run.  It would be good to remove these lines from the list file so this warning doesn't happen.

### LINKED ISSUE #1478: Rework some tests when we switch to a cime version that has nuopc as the default
When we switch to a cime version that has nuopc as the default (cime6.0.8 or later; planned for cesm2_3_beta06), we should make some changes to the test suite:

- [ ] Tests that are currently `_Vnuopc` should be changed to `_Vmct`.
- [ ] The USUMB test uses different testmods for nuopc vs. mct. Make sure each USUMB test is using the correct testmod
- [ ] Change any occurrences of _N2 to _C2 (and other _N# if there are any), except leaving _N2 for any _Vmct tests
- [ ] For tests that are currently failing with the nuopc driver, temporarily add mct versions so that we are testing those configurations until we can get them passing with nuopc.
- [ ] It looks like the `SMS_Ld5_D_P48x1_Vnuopc.f10_f10_mg37.IHistClm51Bgc.izumi_nag.clm-decStart` test can be removed from the expected fails list, and issue #1317 can possibly be closed.

### LINKED ISSUE #1517: Don't allow -namelist as an option in CLM_BLDNML_OPTS
A user on the BB tried to do this and I realized doing so causes so many problems that we really need to disallow it. You end up in UNIX quote hell if you do, and we should spare people from that horrible fate. This is actually why we have CLM_NAMELIST_OPTS in addition to CLM_BLDNML_OPTS in the first place. So if we just disallow having the "-namelist' option in CLM_BLDNML_OPTS and the user told to use CLM_NAMELIST_OPTS in it's place the user is much more likely to be able to do what they need to do rather than have to ask about it. This is one of those places that a naive user can get themselves into real trouble by doing something that seems possible but isn't. So we should just protect against it.

https://bb.cgd.ucar.edu/cesm/threads/issues-setting-up-hybrid-run.5074/#post-35584

### LINKED ISSUE #1523: Change MODEL to COMP_NAME in buildlib
After a particular cime update we need to change MODEL to COMP_NAME in our buildlib. This will make it consistent with the new cime version.

### LINKED ISSUE #1543: f10_f10_musgs grid doesn't work with NUOPC
### Brief summary of bug

The USGS mask doesn't work in NUOPC, so tests with f10_f10_musgs fail when NUOPC is the default driver.

### General bug information

**CTSM version you are using:** ctsm5.1.dev061-37-g6c4fb1ef1 (preliminary ctsm5.1.dev062)

**Does this bug cause significantly incorrect results in the model's science?** No

**Configurations affected:** musgs mask

### Details of bug

The USGS mask wasn't really setup in a consistent way in the system and hasn't been ported over to NUOPC. So the mg37 mask should be used instead.

We realized this was an issue earlier, and switched tests out, but this must have come in after that change.

### Important details of your setup / configuration so we can reproduce the bug

We have two tests on cheyenne that show this problem...

SMS_D_Ld5.f10_f10_musgs.I2000Clm50FatesRs.cheyenne_gnu.clm-FatesColdDef		
SMS_D_Ld5.f10_f10_musgs.I2000Clm50FatesRs.cheyenne_intel.clm-FatesColdDef

There's also one on izumi_nag, and the cheyenne_gnu is in the prebeta test list

### Important output or errors that show the problem

```
Command: /glade/work/erik/ctsm_worktrees/external_updates/cime/scripts/create_newcase --case /glade/scratch/erik/tests_ctsm51d62acl/SMS_D_Ld5.f10_f10_musgs.I2000Clm50FatesRs.cheyenne_gnu.clm-FatesColdDef.GC.ctsm51d62acl_gnu --res f10_f10_musgs --compset I2000Clm50FatesRs --test --machine cheyenne --compiler gnu --project P93300606  --output-root /glade/scratch/erik/tests_ctsm51d62acl  --workflow default --user-mods-dir /glade/work/erik/ctsm_worktrees/external_updates/cime_config/testdefs/testmods_dirs/clm/FatesColdDef  --queue=premium --walltime 00:20:00
Output: Compset longname is 2000_DATM%GSWP3v1_CLM50%FATES_SICE_SOCN_SROF_SGLC_SWAV
Compset specification file is /glade/work/erik/ctsm_worktrees/external_updates/cime_config/config_compsets.xml
Automatically adding SESP to compset
Traceback (most recent call last):
  File "/glade/work/erik/ctsm_worktrees/external_updates/cime/scripts/create_newcase", line 265, in <module>
    _main_func(__doc__)
  File "/glade/work/erik/ctsm_worktrees/external_updates/cime/scripts/create_newcase", line 257, in _main_func
    ngpus_per_node=ngpus_per_node)
  File "/glade/work/erik/ctsm_worktrees/external_updates/cime/scripts/Tools/../../scripts/lib/CIME/case/case.py", line 1795, in create
    ngpus_per_node=ngpus_per_node)
  File "/glade/work/erik/ctsm_worktrees/external_updates/cime/scripts/Tools/../../scripts/lib/CIME/case/case.py", line 974, in configure
    gridinfo = grids.get_grid_info(name=grid_name, compset=self._compsetname, driver=self._comp_interface)
  File "/glade/work/erik/ctsm_worktrees/external_updates/cime/scripts/Tools/../../scripts/lib/CIME/XML/grids.py", line 80, in get_grid_info
    domains  = self._get_domains(component_grids, atmlevregex, lndlevregex, driver)
  File "/glade/work/erik/ctsm_worktrees/external_updates/cime/scripts/Tools/../../scripts/lib/CIME/XML/grids.py", line 219, in _get_domains
    mask_domain_node = self.get_optional_child("domain", attributes={"name":domains["MASK_GRID"]},
KeyError: 'MASK_GRID'
```

### LINKED ISSUE #1544: SSP cases fail with mosart1_0_44
### Brief summary of bug

mosart1_0_44 has a change where MOSART_MODE is modified from inside of buildnml. This breaks the SSP tests as some of the cases are setup in readonly mode.

### General bug information

**CTSM version you are using:** ctsm5.1.dev061-39-g9d84f7788

**Does this bug cause significantly incorrect results in the model's science?** No

**Configurations affected:** SSP tests

### Details of bug

Tests that fail:

SSP_D_Ld10.f19_g17.I1850Clm51Bgc.cheyenne_intel.clm-rtmColdSSP		
SSP_D_Ld4.f09_g17.I1850Clm50BgcCrop.cheyenne_intel.clm-ciso_rtmColdSSP		
SSP_Ld10.f19_g17.I1850Clm50Bgc.cheyenne_intel.clm-rtmColdSSP

### Important output or errors that show the problem

```
 ---------------------------------------------------
2021-11-08 20:16:59: ERROR: Cannot modify case, read_only. Case must be opened with read_only=False and can only be modified within a context manager
 ---------------------------------------------------
```

### LINKED ISSUE #1545: NUOPC tests failing with floating point exceptions
### Brief summary of bug

The following tests fail in ctsm5.1.dev061 with floating point exceptions

ERS_Vnuopc_D_Mmpi-serial_Ld5.1x1_brazil.I2000Clm50FatesRs.cheyenne_gnu.clm-FatesColdDef		
ERS_Vnuopc_Ly5_P72x1.f10_f10_mg37.IHistClm45BgcCrop.cheyenne_intel.clm-cropMonthOutput		
SMS_Vnuopc_D_Ld5.f10_f10_mg37.I2000Clm50FatesRs.cheyenne_gnu.clm-FatesColdDef		
SMS_Vnuopc_D_Mmpi-serial_Ld5.5x5_amazon.I2000Clm50FatesRs.cheyenne_gnu.clm-FatesColdDef	

### General bug information

**CTSM version you are using:** ctsm5.1.dev061

**Does this bug cause significantly incorrect results in the model's science?** No

**Configurations affected:** Some lower resolution Fates Clm50 and a Clm45BgcCrop monthly case

### Details of bug

The Clm45BgcCrop issue seems to be a NaN coming in an irrigation field from RTM.

The FATES issues seem to be due to bad data (likely a NaN) in FATES history output.

### Important details of your setup / configuration so we can reproduce the bug

Note, these tests (other than the third one, passed for @billsacks in ctsm5.1.dev054

### Important output or errors that show the problem

The FATES issues show this in the traceback in cesm.log

SMS_Vnuopc_D_Ld5.f10_f10_mg37.I2000Clm50FatesRs.cheyenne_gnu.clm-FatesColdDef
```
71:
0:
0:Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation.
0:
0:Backtrace for this error:
0:#0  0x2acb4539daff in ???
0:#1  0x5464ff in __ncdio_pio_MOD_ncd_io_1d_double
0:      at /glade/scratch/erik/ctsm5.1.dev061/src/main/ncdio_pio.F90.in:1808
0:#2  0x60b8ce in hfields_write
0:      at /glade/scratch/erik/ctsm5.1.dev061/src/main/histFileMod.F90:3545
0:#3  0x62fa49 in __histfilemod_MOD_hist_htapes_wrapup
0:      at /glade/scratch/erik/ctsm5.1.dev061/src/main/histFileMod.F90:4065
0:#4  0x5624b4 in __clm_driver_MOD_clm_drv
0:      at /glade/scratch/erik/ctsm5.1.dev061/src/main/clm_driver.F90:1405
0:#5  0x517798 in modeladvance
0:      at /glade/scratch/erik/ctsm5.1.dev061/src/cpl/nuopc/lnd_comp_nuopc.F90:885
```

The Clm45BgcCrop case shows that Flrl_irrig is NaN in RTM after running to 18500829

### LINKED ISSUE #1557: ctsm_sci tests for f09 and f19 need more processors on izumi
### Brief summary of bug

Some of the f09 and f19 tests in the ctsm_sci test list fail because they need more processors.

### General bug information

**CTSM version you are using:** ctsm5.1.dev061
**Does this bug cause significantly incorrect results in the model's science?** No

**Configurations affected:** 

SMS_Ld3.f09_g17.I1850Clm50BgcCropCru.izumi_intel.clm-default

### Details of bug

Using 2 or 4 nodes fails for several ctsm_sci cases on izumi.

### Important details of your setup / configuration so we can reproduce the bug

These tests either fail because of too little time or abort with an error because they need more processors.

SMS_Ld5.f09_g17.I1850Clm45Bgc.izumi_intel.clm-default		
SMS_Ld5.f09_g17.I1850Clm45BgcCru.izumi_intel.clm-default		
SMS_Ld5.f09_g17.IHistClm45Sp.izumi_intel.clm-default		
SMS_Ld5.f19_g17.I1850Clm50Bgc.izumi_intel.clm-default		
SMS_Ld5.f09_g17.I1850Clm50Bgc.izumi_intel.clm-default		
SMS_Ld5.f09_g17.IHistClm45Bgc.izumi_intel.clm-default		
SMS_Ld5.f09_g17.IHistClm45BgcCru.izumi_intel.clm-default		
SMS_Ld5.f09_g17.IHistClm50BgcCrop.izumi_intel.clm-default		
SMS_Ld5.f09_g17.IHistClm50Sp.izumi_intel.clm-default		
SMS_Ld5.f09_g17.IHistClm50SpCru.izumi_intel.clm-default		
SMS_Ld5.f19_g17.IHistClm50BgcCrop.izumi_intel.clm-default

### Important output or errors that show the problem

cesm.log dies as follows

```
[150] forrtl: severe (174): SIGSEGV, segmentation fault occurred
[150] Image              PC                Routine            Line        Source             
[150] cesm.exe           00000000011DCDEA  Unknown               Unknown  Unknown
[150] libpthread-2.17.s  00007FE2BAB4F5F0  Unknown               Unknown  Unknown
[150] cesm.exe           00000000006F6F90  cnfunmod_mp_cnfun         725  CNFUNMod.F90
[150] cesm.exe           0000000000A1FB07  soilbiogeochemcom         737  SoilBiogeochemCompetitionMod.F90
[150] cesm.exe           0000000000C9AF12  cndrivermod_mp_cn         398  CNDriverMod.F90
[150] cesm.exe           0000000000790652  cnvegetationfacad         953  CNVegetationFacade.F90
[150] cesm.exe           0000000000555BDD  clm_driver_mp_clm         983  clm_driver.F90
[150] cesm.exe           000000000050C0CF  lnd_comp_mct_mp_l         451  lnd_comp_mct.F90
[150] cesm.exe           000000000043BAC4  component_mod_mp_         737  component_mod.F90
[150] cesm.exe           000000000041AC14  cime_comp_mod_mp_        2855  cime_comp_mod.F90
[150] cesm.exe           000000000043B700  MAIN__                    153  cime_driver.F90
[150] cesm.exe           0000000000418422  Unknown               Unknown  Unknown
[150] libc-2.17.so       00007FE2BA794505  __libc_start_main     Unknown  Unknown
[150] cesm.exe           0000000000418329  Unknown               Unknown  Unknown
```

### PR DESCRIPTION
**Pull request by ekluzek**
### Description of changes

Update to the externals in cesm2_3_beta06 which makes the NUOPC driver the default (rather than MCT).

### Specific notes

Contributors other than yourself, if any: @mvertens 

CTSM Issues Fixed (include github issue #):
 Fixes #1557
 Fixes #1478
 Fixes #1365
 Fixes #1083
 Fixes #1237
 Fixes #1502
 Fixes #1476
 Fixes #1517
 Fixes #1523
 Fixes #1545
 Fixes #1192
 Fixes #1544
 Fixes #1543

Are answers expected to change (and if so in what way)? Yes! (as compared to simulations with the MCT driver)
  Answers are different, but not climate-changing. The SSC has vetted the changes.
  Answers should not change compared to simulations with the same driver for the previous version

Any User Interface Changes (namelist or namelist defaults changes)? Yes!
  The datm namelist and streams handling changes significantly
  New user_nl_datm_streams file to modify datm streams (rather than the stream specific user_datm.stream.txt.* files)
  Mesh files are required for grids
  domain files no longer used

Testing performed, if any:  will be regular
  Waiting for the aux_clm testing to be run on the cesm2_3_beta06 tag

### CODE REVIEW

**Comment by ekluzek** [comment]
Will also need to update externals for: cism, mosart, rtm with updated test lists that change nuopc tests to mct.

**Comment by ekluzek** [comment]
@olyson you ran some simulations with NUOPC to show that the climate is the same. Can you point me to those simulations and diagnostics so I can add it to the ChangeLog?

**Comment by olyson** [comment]
The MCT and NUOPC historical case directories are here:

/glade/work/oleson/cesm2_3_alpha03a/cime/scripts/ctsm51_cesm23a03a_1deg_GSWP3V1_mct_hist
/glade/work/oleson/cesm2_3_alpha03a/cime/scripts/ctsm51_cesm23a03a_1deg_GSWP3V1_nuopc_hist

The history output is here (for now):

/glade/p/cgd/tss/people/oleson/ctsm51_cesm23a03a_1deg_GSWP3V1_mct_hist
/glade/p/cgd/tss/people/oleson/ctsm51_cesm23a03a_1deg_GSWP3V1_nuopc_hist

The diagnostics comparing NUOPC to MCT are here:

https://webext.cgd.ucar.edu/I20TR/ctsm51_cesm23a03a_1deg_GSWP3V1_nuopc_hist/lnd/ctsm51_cesm23a03a_1deg_GSWP3V1_nuopc_hist.1995_2014-ctsm51_cesm23a03a_1deg_GSWP3V1_mct_hist.1995_2014/setsIndex.html

**Comment by ekluzek** [comment]
I'm getting a few unexpected fails, with floating point trap exceptions.

ERS_D_Mmpi-serial_Ld5.1x1_brazil.I2000Clm50FatesRs.cheyenne_gnu.clm-FatesColdDef		
ERS_Ly5_P72x1.f10_f10_mg37.IHistClm45BgcCrop.cheyenne_intel.clm-cropMonthOutput		
SMS_D_Ld5.f10_f10_mg37.I2000Clm50FatesRs.cheyenne_gnu.clm-FatesColdDef			
SMS_D_Mmpi-serial_Ld5.5x5_amazon.I2000Clm50FatesRs.cheyenne_gnu.clm-FatesColdDef

**Comment by billsacks** [comment]
@ekluzek for what it's worth, I'm pretty sure those tests were *not* failing when I ran them with the beta06 externals in September: https://github.com/ESCOMP/CTSM/issues/1478#issuecomment-912674171

**Comment by ekluzek** [comment]
Yes, looks like they worked then. Although SMS_D_Ld5.f10_f10_mg37.I2000Clm50FatesRs.cheyenne_gnu.clm-FatesColdDef is a new test since then. I'm running them in ctsm5.1.dev061 with NUOPC to see if they fail there with older externals. It could be something in an externals update (including updates fates).

**Comment by billsacks** [comment]
@ekluzek see https://github.com/ekluzek/CTSM/pull/2 for a fix for the LILAC test. This doesn't address the issue with a user-defined machine, but gets the LILACSMOKE test passing.

**Comment by billsacks** [comment]
> ERS_D_Mmpi-serial_Ld5.1x1_brazil.I2000Clm50FatesRs.cheyenne_gnu.clm-FatesColdDef
> ERS_Ly5_P72x1.f10_f10_mg37.IHistClm45BgcCrop.cheyenne_intel.clm-cropMonthOutput
> SMS_D_Ld5.f10_f10_mg37.I2000Clm50FatesRs.cheyenne_gnu.clm-FatesColdDef
> SMS_D_Mmpi-serial_Ld5.5x5_amazon.I2000Clm50FatesRs.cheyenne_gnu.clm-FatesColdDef

@ekluzek @glemieux - Just following up from the conversation we just had: it actually looks like all of the cheyenne_gnu fates tests are failing (at least based on a fairly recent test list) (the one not listed here is the musgs test that you have removed). At a glance, it looks like there are equivalent intel tests for these; given that, I am comfortable with these being expected failures if that's what you prefer, since the equivalent configuration is already covered for the sake of baseline comparisons.

**Comment by glemieux** [comment]
https://github.com/NGEET/fates/pull/807 should address the failing gnu debug `FatesColdDef` tests above.   The PR shows that the fix runs B4B for the `fates` suite against the `fates-sci.1.49.0_api.17.0.0-ctsm5.1.dev061` baseline.  Given that the PR is developed off of a more recent, answer-changing fates tag, DIFFs should be expected in comparing against the original `ctsm5.1.dev061` baseline.

That said, I ran a series of comparison tests against `ctsm5.1.dev061` using a fates branch that had fixes applied from fates PR 807 to `fates-sci.1.48.0_api.17.0.0` to assess the affect of the upcoming `ctsm5.1.dev062` update.  I ran the `5x5_amazon` test noted above in the following configurations of ctsm tag/branch and test prefix modification:

1. Upcoming `ctsm5.1.dev062` branch, `amazon` test  _without_ a `Vnuopc` prefix: DIFFs
2. Upcoming `ctsm5.1.dev062` branch, `amazon` test  _with_ a `Vnuopc` prefix: FIELDLIST only DIFFs
3. `ctsm5.1.dev061`, `amazon` test _without_ a `Vnuopc` prefix: B4B

(Note that to run the comparison test for 2. I generated a new `ctsm5.1.dev061` baseline using this same fates branch for the `Vnuopc` test which can be found here: `/glade/u/home/glemieux/scratch/ctsm-tests/tests_nuopc-gendev061-amazon-sci148`.  This idea was inspired by @ekluzek testing in #1545.)

So the nuopc update is generating DIFFs, not unexpectedly.  The thing that is interesting to me is that the addition of the `Vnuopc` prefix to the test changes the scope of the DIFFs.  I'm not familiar with how the prefix changes the test setup, but I thought perhaps it might narrow down the source of the diffs?  Additionally, looking at the RMS results, they seem reasonable, but I saw a change in `time_bounds` which I hadn't seen before.

The test run cases can be found here:
1. `/glade/u/home/glemieux/scratch/ctsm-tests/tests_nuopc-sci148_mod-sms-amazon-fix-comp`
2. `/glade/u/home/glemieux/scratch/ctsm-tests/tests_nuopc-prbranch-amazon-sci148-comp`
3. `/glade/u/home/glemieux/scratch/ctsm-tests/tests_nuopc-dev061-sci148_mod-sms-amazon-fix-comp`

File: README.NUOPC_driver [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+streams files.
+Full ESMF Library is used:
+The full ESMF Library is used and required to be built in order to run the model.
+Single Point cases:
+Single point cases can now set their location using PTS_LAT and PTS_LON.
+
+What's removed?
+
+Domain files are no longer used. And mapping for regriding is created on the fly
+rather than using fixed mapping files.
```
[the remark this answers is not in the document]
Comment (ekluzek) [reply README.NUOPC_driver]: Ok, got that update.

**Comment by ekluzek** [comment]
@billsacks and @mvertens can you comment on my commit 04456d7c08535164d34d49a667ae3f8f1d841246 that got the RTM test to work? It's not initializing the irrig field to spval like the other fields. The setting to spval makes sense to me in terms of being able to write history output and call them missing value. But, RTM or CTSM must be doing something bad with the spval data that ends up with a NaN at some point. So this gets the model to run, but I'm not sure it's the right long term solution that will allow irrig to be written out to history. Anyway, I'd like to hear both your takes on what you think the long term solution should be.

**Comment by billsacks** [comment]
@ekluzek - see https://github.com/ESCOMP/CTSM/issues/1528 . My sense, like yours, is that the long-term solution is to set everything to spval over ocean points. But we need to determine why that leads to issues for some fields.

**Comment by ekluzek** [comment]
So the test that failed was ERP_D_Ld5.f19_g17.I2000Clm50BgcCru.cheyenne_intel.clm-default. But, it worked on resending it. So this perhaps indicates there might be some rare random problems where rerunning works fine.

**Comment by ekluzek** [comment]
The error was all high level ESMF stuff as follows...

```
956:MPT: 0x00002b5ea58506da in waitpid () from /glade/u/apps/ch/os/lib64/libpthread.so.0
956:MPT: Missing separate debuginfos, use: zypper install glibc-debuginfo-2.22-49.16.x86_64
956:MPT: (gdb) #0  0x00002b5ea58506da in waitpid ()
956:MPT:    from /glade/u/apps/ch/os/lib64/libpthread.so.0
956:MPT: #1  0x00002b5ea6521306 in mpi_sgi_system (
956:MPT: #2  MPI_SGI_stacktraceback (
956:MPT:     header=header@entry=0x7fffc4219fd0 "MPT ERROR: Rank 956(g:956) received signal SIGSEGV(11).\n\tProcess ID: 30933, Host: r12i2n22, Program: /glade/scratch/erik/tests_ctsm51d62acl/ERP_D_Ld5.f19_g17.I2000Clm50BgcCru.cheyenne_intel.clm-defaul"...) at sig.c:340
956:MPT: #3  0x00002b5ea65214ff in first_arriver_handler (signo=signo@entry=11, 
956:MPT:     stack_trace_sem=stack_trace_sem@entry=0x2b5eb5e20080) at sig.c:489
956:MPT: #4  0x00002b5ea6521793 in slave_sig_handler (signo=11, 
956:MPT:     siginfo=<optimized out>, extra=<optimized out>) at sig.c:565
956:MPT: #5  <signal handler called>
956:MPT: #6  0x00002b5e9f9a4a14 in ESMC_Base::ESMC_BaseGetStatus (this=0xffffffff)
956:MPT:     at /glade/p/cesmdata/cseg/PROGS/build/28560/esmf-8.2.0b23/src/include/ESMCI_Base.h:163
956:MPT: #7  0x00002b5e9fe04094 in ESMCI::DistGrid::getRegDecomp (this=0xffffffff)
956:MPT:     at /glade/p/cesmdata/cseg/PROGS/build/28560/esmf-8.2.0b23/src/Infrastructure/DistGrid/interface/../include/ESMCI_DistGrid.h:277
956:MPT: #8  0x00002b5e9fe0000b in c_esmc_distgridget_ (ptr=0xffff00009960, 
956:MPT:     dimCount=0x0, tileCount=0x0, deCount=0x0, localDeCount=0x0, 
956:MPT:     minIndexPDimPTile=0x0, maxIndexPDimPTile=0x7fffc421ad98, 
956:MPT:     elementCountPTile=0x7fffc421ad98, elementCountPTileI8=0x7fffc421aaa0, 
956:MPT:     minIndexPDimPDe=0x2b5e9ff00714 <std::vector<std::string, std::allocator<std::string> >::max_size() const+40>, maxIndexPDimPDe=0x7fffc421ad98, 
956:MPT:     elementCountPDe=0x7fffc421ad98, elementCountPDeI8=0x1fffffffffffffff, 
956:MPT:     localDeToDeMap=0x1fffffffffffffff, tileListPDe=0x1fffffffffffffff, 
956:MPT:     indexCountPDimPDe=0x0, collocationPDim=0x7fffc421ad98, regDecompFlag=0x1, 
956:MPT:     connectionCount=0x7fffc421ab30, 
956:MPT:     connectionList=0x2b5e9ff00a9e <std::vector<std::string, std::allocator<std::string> >::_M_check_len(unsigned long, char const*) const+242>, 
956:MPT:     indexTK=0x7fffc421ad98, indexflag=0x1, delayout=0x2b5ea2bfeacc, 
956:MPT:     rc=0x1fffffffffffffff)
956:MPT:     at /glade/p/cesmdata/cseg/PROGS/build/28560/esmf-8.2.0b23/src/Infrastructure/DistGrid/interface/ESMCI_DistGrid_F.C:346
```

The backtrace goes all the way up 54 levels and it's all in ESMF code. It looks like the problem is a floating point exception. Here's the end of the traceback...

```
956:MPT: -----stack traceback ends-----
956:MPT: On host r12i2n22, Program /glade/scratch/erik/tests_ctsm51d62acl/ERP_D_Ld5.f19_g17.I2000Clm50BgcCru.cheyenne_intel.clm-default.GC.ctsm51d62acl_int/bld/cesm.exe, Rank 956, Process 30933: Dumping core on signal SIGSEGV(11) into directory /glade/scratch/erik/tests_ctsm51d62acl/ERP_D_Ld5.f19_g17.I2000Clm50BgcCru.cheyenne_intel.clm-default.GC.ctsm51d62acl_int/run
-1:MPT ERROR: MPI_COMM_WORLD rank 956 has terminated without calling MPI_Finalize()
-1:	aborting job
MPT: Received signal 11
```

**Comment by ekluzek** [comment]
Another thing I checked was to make sure there are still some explicit MCT tests for prealpha and prebeta testing and there are. 

prealpha
ERP_D_Ld3_Vmct" grid="f09_g17" compset="I2000Clm50Sp" testmods="clm/prescribed
SMS_D_Ld1_Mmpi-serial_Vmct" grid="f45_f45_mg37" compset="I2000Clm50SpRs" testmods="clm/ptsRLA
DAE_N2_D_Lh12_Vmct" grid="f10_f10_mg37" compset="I2000Clm50BgcCrop" testmods="clm/DA_multidrv

prebeta:
ERS_D_Ld5_Vmct" grid="f10_f10_mg37" compset="I2000Clm50BgcCropRtm" testmods="rtm/rtmOnFloodOnEffvelOn
SMS_D_Vmct_Lm1_Mmpi-serial" grid="CLM_USRDAT" compset="I1PtClm50SpRs" testmods="clm/USUMB_mct

The total list of MCT tests are:

``` xml
  <test name="ERP_D_Ld3_Vmct" grid="f09_g17" compset="I2000Clm50Sp" testmods="clm/prescribed">
  <test name="ERI_D_Ld9_P48x1_Vmct" grid="f10_f10_mg37" compset="I2000Clm50Sp" testmods="clm/SNICARFRC">
  <test name="SMS_D_Ln9_P480x3_Vmct" grid="f19_g17" compset="IHistClm50Sp" testmods="clm/waccmx_offline">
  <test name="ERP_D_Ld10_P36x2_Vmct" grid="f10_f10_mg37" compset="IHistClm51BgcCrop" testmods="clm/ciso_decStart">
  <test name="ERS_D_Ld5_Vmct" grid="f10_f10_mg37" compset="I2000Clm50BgcCropRtm" testmods="rtm/rtmOnFloodOnEffvelOn">
  <test name="ERP_P72x2_Lm7_Vmct" grid="f10_f10_mg37" compset="I2000Clm50BgcCrop" testmods="clm/irrig_alternate_monthly">
  <test name="ERS_Ly3_P72x2_Vmct" grid="f10_f10_mg37" compset="IHistClm50BgcCropG" testmods="clm/cropMonthOutput">
  <test name="SMS_Vmct" grid="f10_f10_mg37" compset="I2000Clm50BgcCrop" testmods="clm/crop">
  <test name="SMS_Vmct_D_Ld1" grid="f09_g17" compset="I1850Clm50BgcSpinup" testmods="clm/cplhist">
  <test name="SMS_D_Ld1_Mmpi-serial_Vmct" grid="f45_f45_mg37" compset="I2000Clm50SpRs" testmods="clm/ptsRLA">
  <test name="SMS_D_Ld1_P48x1_Vmct" grid="f10_f10_mg37" compset="I2000Clm50BgcCru" testmods="clm/af_bias_v7">
  <test name="DAE_N2_D_Lh12_Vmct" grid="f10_f10_mg37" compset="I2000Clm50BgcCrop" testmods="clm/DA_multidrv">
  <test name="SMS_D_Vmct_Lm1_Mmpi-serial" grid="CLM_USRDAT" compset="I1PtClm50SpRs" testmods="clm/USUMB_mct">
  <test name="ERS_D_Mmpi-serial_Ld5_Vmct" grid="1x1_brazil" compset="I2000Clm50FatesRs" testmods="clm/FatesColdDef">
```

**Pull request merged**